container-search/src/main/java/com/yahoo/search/Query.java

```diff
@@ -970,10 +971,12 @@ private void copyPropertiesTo(Query clone) {
      * @param create if true this is created if not already set
      * @return the session id of this query, or null if not set and create is false
      */
-    public SessionId getSessionId(boolean create) {
+    public Utf8String getSessionId(String rankProfile, boolean create) {
```

Comment:
... that also avoids having to know at three different places how to supply the right rankProfile parameter.